client/my-sites/domains/domain-management/dataviews/use-actions.tsx

```diff
@@ -0,0 +1,259 @@
+import { Icon, drawerLeft, info, update } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import { navigate } from 'calypso/lib/navigate';
+import { domainManagementEditContactInfo } from '../../paths';
+import { AutoRenewDiolog } from './components/auto-renew-dialog';
+import { useDomainsDataViewsContext } from './use-context';
+
+export function useActions( { sidebarMode }: { sidebarMode?: boolean }, onClose?: () => void ) {
+	const translate = useTranslate();
+	const {
+		getFullDomain,
+		sites,
+		getSiteSlug,
+		isAllSitesView,
+		domainStatusPurchaseActions,
+		updatingDomain,
+		userCanSetPrimaryDomains = false,
+		onDomainAction,
+		handleUpdateContactInfo,
+	} = useDomainsDataViewsContext();
+
+	const actions: Action< PartialDomainData >[] = [
+		{
+			id: 'manage-domain',
+			isPrimary: true,
+			icon: drawerLeft,
+			label: ( domains: Array< PartialDomainData > ) => {
+				const domain = domains.length > 0 && domains[ 0 ] && getFullDomain( domains[ 0 ] );
+				return domain && domain.type === domainTypes.TRANSFER
+					? translate( 'View transfer' )
+					: translate( 'View settings' );
+			},
+			callback: ( domains: Array< PartialDomainData > ) => {
+				const domain = domains.length > 0 && domains[ 0 ] && getFullDomain( domains[ 0 ] );
+				if ( ! domain ) {
+					return;
+				}
+				const url = domainManagementLink( domain, getSiteSlug( domains[ 0 ] ), isAllSitesView );
+				navigate( url );
+			},
+			isEligible( partialDomain: PartialDomainData ) {
+				const domain = getFullDomain( partialDomain );
+				if ( ! domain ) {
+					return false;
+				}
+				return domain.type !== domainTypes.WPCOM;
+			},
+		},
+		{
+			id: 'manage-dns-settings',
+			callback: ( domains: Array< PartialDomainData > ) => {
+				const domain = domains.length > 0 && domains[ 0 ] && getFullDomain( domains[ 0 ] );
+				if ( domain ) {
+					onDomainAction?.( 'manage-dns-settings', domain );
+					const url = domainManagementDNS( getSiteSlug( domains[ 0 ] ), domain.domain );
+					navigate( url );
+				}
+			},
+			label: translate( 'Manage DNS' ),
+			supportsBulk: false,
+			isEligible( partialDomain: PartialDomainData ) {
+				const domain = getFullDomain( partialDomain );
+				if ( ! domain ) {
+					return false;
+				}
+				return (
+					domain.canManageDnsRecords &&
+					domain.transferStatus !== transferStatus.PENDING_ASYNC &&
+					domain.type !== domainTypes.SITE_REDIRECT
+				);
+			},
+		},
+		{
+			id: 'manage-contact-info',
+			icon: <Icon icon={ info } />,
+			callback: ( domains: Array< PartialDomainData > ) => {
+				if ( domains.length === 0 ) {
+					return;
+				}
+				if ( domains.length === 1 ) {
+					const domain = domains[ 0 ];
+					const url = domainManagementEditContactInfo( getSiteSlug( domain ), domain.domain );
+					navigate( url );
+				} else {
+					handleUpdateContactInfo( domains );
+				}
+			},
+			label: translate( 'Manage contact information' ),
+			supportsBulk: ! sidebarMode,
```

Comment:
I understand this is trying to enable/disable bulk support depending on what layout is displayed (table or list)? We don't need that, the layouts themselves handle that gracefully (list layout doesn't support bulk actions yet).

Reply:
Fixed here: https://github.com/Automattic/wp-calypso/pull/97677/commits/1244ab18e214695fdc68b1a5d58277a67b676f25